chore: add controller-gen to tasks list and some order

sxd opened the pull request:
The manifests wasn't a dependency for the uncommitted changes and
we added the controller-gen missing task, now the uncommitted has as
dependency the manifest and manifest controller-gen

sxd committed:
ci: add controller-gen task

Add a task to run controller-gen, and put it as a dependency for
the uncommitted and manifest task.

Signed-off-by: Jonathan Gonzalez V. <jonathan.gonzalez@enterprisedb.com>
Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>

jbattiato commented:
The failing CI check is caused by the task `manifest` which creates artifacts in the repo directory making the `uncommitted` check fails:

```
$ task uncommitted
task: Task "controller-gen" is up to date
task: [manifest] dagger -s call -m github.com/sagikazarmark/daggerverse/kustomize@${DAGGER_KUSTOMIZE_SHA} edit --source config --dir default set image --image controller=ghcr.io/pppa-testing:dev/add_controller-gen directory export --path manifest-build

./plugin-barman-cloud/manifest-build
task: [manifest] dagger -s call -m github.com/sagikazarmark/daggerverse/kustomize@${DAGGER_KUSTOMIZE_SHA} build --source manifest-build --dir default export --path manifest.yaml

./plugin-barman-cloud/manifest.yaml
task: [uncommitted] GITHUB_REF= dagger -s call -m github.com/cloudnative-pg/daggerverse/uncommitted@${DAGGER_UNCOMMITTED_SHA} check-uncommitted --source . stdout
Error: response from query: input: container.from.withExec.withExec.withMountedDirectory.withWorkdir.withExec.stdout resolve: process "check_uncommitted_git_changes" did not complete successfully: exit code: 1

Stderr:
INFO checking for uncommitted git changes
ERROR found uncommitted changes
task: Failed to run task "uncommitted": exit status 1

$ git status --porcelain 
?? manifest-build/
?? manifest.yaml
```

fcanovai committed:
fix: obsolete deepcopy

Having automated controller-gen invocation, we noticed that the
generated deepcopy file was obsolete.

Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
